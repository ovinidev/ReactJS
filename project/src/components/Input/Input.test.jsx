import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Input } from '.';

describe('<Posts />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<Input handleChange={fn} searchValue="testando" />);

    const input = screen.getByPlaceholderText(/Faça sua busca/i);

    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<Input handleChange={fn} searchValue="um valor qualquer" />);

    const input = screen.getByPlaceholderText(/Faça sua busca/i);
    const valueInput = 'valor';

    userEvent.type(input, valueInput);

    expect(input.value).toBe('um valor qualquer');
    expect(fn).toHaveBeenCalledTimes(valueInput.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<Input handleChange={fn} searchValue="testando" />);

    expect(container).toMatchSnapshot();
  });
});
