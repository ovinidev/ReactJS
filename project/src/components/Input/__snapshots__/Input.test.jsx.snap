// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div>
  <input
    placeholder="FaÃ§a sua busca"
    type="search"
    value=""
  />
</div>
`;
